tools/releaser/main.go

```diff
@@ -0,0 +1,304 @@
+	if verbose {
+		fmt.Println("Parsing temporary bzl file to prepare deps.bzl and WORKSPACE modifications.")
+	}
+	maybeRules, workspaceDirectives, err := readFromTmp(tmpBzlPath)
+	if err != nil {
+		return err
+	}
+
+	// update deps
+	if verbose {
+		fmt.Println("Writing new deps.bzl")
+	}
+	if err := updateDepsBzlWithRules(depsPath, maybeRules); err != nil {
+		return err
+	}
+
+	// append WORKSPACE with directives at the end.
+	// except we cannot append directly because the earlier bazel //:gazelle run modified WORKSPACE
+	// so we truncate and seek to the beginning again before writing all of what we want
+	if verbose {
+		fmt.Println("Append WORKSPACE with directives")
+	}
+	_ /* new offset */, err = workspace.Seek(0, os.SEEK_SET)
+	if err != nil {
+		return err
+	}
+
+	// write the directive-less workspace and update repos
+	if _, err := workspace.Write(workspaceWithoutDirectives); err != nil {
+		return err
+	}
+	if _, err := workspace.Write(workspaceDirectives); err != nil {
+		return err
+	}
+
+	// cleanup before final gazelle run
+	//
+	// note that we also have a defer for os.Remove so it gets cleaned up if there are earlier errors.
+	// This defer will throw an error from this point on, but we're swallowing it anyways.
+	if verbose {
+		fmt.Println("Cleaning up temporary files")
+	}
+	if err := os.Remove(tmpBzlPath); err != nil {
+		return err
+	}
+
+	if verbose {
+		fmt.Println("Running final gazelle run, and copying some language specific build files.")
+	}
+	finalizationCommands := []struct {
+		cmd  string
+		args []string
+		then func() error
+	}{
+		{cmd: "bazel", args: []string{"run", "//:gazelle"}},
+		{cmd: "bazel", args: []string{"build", "//language/proto:known_imports"}, then: copyClosure(
+			path.Join(os.Getenv("BINDIR"), "language/proto/known_imports.go"), /* src */
+			"language/proto/known_imports.go",
+		)},
+		{cmd: "bazel", args: []string{"build", "//language/proto:known_proto_imports"}, then: copyClosure(
+			path.Join(os.Getenv("BINDIR"), "language/proto/known_proto_imports.go"), /* src */
+			"language/proto/known_proto_imports.go",
+		)},
+		{cmd: "bazel", args: []string{"build", "//language/proto:known_go_imports"}, then: copyClosure(
+			path.Join(os.Getenv("BINDIR"), "language/proto/known_go_imports.go"), /* src */
+			"language/proto/known_go_imports.go",
+		)},
+	}
+	for _, c := range finalizationCommands {
+		cmd := exec.CommandContext(ctx, c.cmd, c.args...)
+		cmd.Dir = os.Getenv("BUILD_WORKSPACE_DIRECTORY")
+		if out, err := cmd.CombinedOutput(); err != nil {
+			fmt.Println(string(out))
+			if c.then != nil {
+				if err := c.then(); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	if verbose {
+		fmt.Println("Release prepared.")
+	}
+	return nil
+}
+
+func copyClosure(srcPath, destPath string) func() error {
+	return func() error {
+		src, err := os.Open(srcPath)
+		if err != nil {
+			return err
+		}
+		dest, err := os.OpenFile(destPath, os.O_WRONLY, 0644)
```

Comment:
```suggestion
		dest, err := os.Create(destPath)
```
a bit simpler this way, when opening for write only